server/parsec/components/shamir.py

```diff
@@ -0,0 +1,195 @@
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepInvalidRecipient()
+                case ShamirAddRecoverySetupStoreBadOutcome.ALREADY_SET:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepShamirSetupAlreadyExists()
+
+                case ShamirAddRecoverySetupValidateBadOutcome.BRIEF_INVALID_DATA:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepBriefInvalidData()
+                case ShamirAddRecoverySetupValidateBadOutcome.SHARE_INVALID_DATA:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepShareInvalidData()
+                case ShamirAddRecoverySetupValidateBadOutcome.SHARE_RECIPIENT_NOT_IN_BRIEF:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepShareRecipientNotInBrief()
+                case ShamirAddRecoverySetupValidateBadOutcome.DUPLICATE_SHARE_FOR_RECIPIENT:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepDuplicateShareForRecipient()
+                case ShamirAddRecoverySetupValidateBadOutcome.AUTHOR_INCLUDED_AS_RECIPIENT:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepAuthorIncludedAsRecipient()
+                case ShamirAddRecoverySetupValidateBadOutcome.MISSING_SHARE_FOR_RECIPIENT:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepMissingShareForRecipient()
+                case ShamirAddRecoverySetupValidateBadOutcome.THRESHOLD_GREATER_THAN_TOTAL_SHARES:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepThresholdGreaterThanTotalShares()
+                case ShamirAddRecoverySetupValidateBadOutcome.SHARE_INCOHERENT_TIMESTAMP:
+                    return authenticated_cmds.latest.shamir_recovery_setup.RepShareIncoherentTimestamp()
+
+                case TimestampOutOfBallpark() as error:
+                    return (
+                        authenticated_cmds.latest.shamir_recovery_setup.RepTimestampOutOfBallpark(
+                            server_timestamp=error.server_timestamp,
+                            client_timestamp=error.client_timestamp,
+                            ballpark_client_early_offset=error.ballpark_client_early_offset,
+                            ballpark_client_late_offset=error.ballpark_client_late_offset,
+                        )
+                    )
+                case RequireGreaterTimestamp() as error:
+                    return (
+                        authenticated_cmds.latest.shamir_recovery_setup.RepRequireGreaterTimestamp(
+                            strictly_greater_than=error.strictly_greater_than
+                        )
+                    )
+
+    async def remove_recovery_setup(
+        self,
+        organization_id: OrganizationID,
+        author: UserID,
+    ) -> (
+        None
+        | ShamirAddRecoverySetupStoreBadOutcome
```

Comment:
Return type differs here and in the memory implementation, I guess you should remove the errors that are not actually used from here

Reply:
Also you should introduce a `ShamirRemoveRecoverySetupStoreBadOutcome` instead of using `ShamirAddRecoverySetupStoreBadOutcome` in `remove_recovery_setup` (to stays consistent with the rest of the code base)

Alternatively, you could introduce a special error (named something like `ShamirAddOrRemoveRecoverySetupStoreBadOutcome`) to be used by both. The point being to have an error with a name making it clear it is used in multiple places (and hence care must be taken if another error case is added to this error type)